tests/run-make/asm-incr-cache/rmake.rs

```diff
@@ -0,0 +1,16 @@
+extern crate run_make_support;
```

Comment:
Could you add a comment and some references for this test? I think you added in the other PR which depends on this to get merged first?

```
// rustc should be able to emit required files (asm, llvm-*, etc) during incremental
// compilation on the first pass by running the code gen as well as on subsequent runs -
// extracting them from the cache
//
// Fixes: rust-lang/rust#89149
// Fixes: rust-lang/rust#88829
// Also see discussion at <https://internals.rust-lang.org/t/interaction-between-incremental-compilation-and-emit/20551>.
```

This helps anyone who's trying to debug test failures out a ton.